modules/hardware/common/shared-mem.nix

```diff
@@ -51,6 +108,99 @@ in
         else
           value;
     };
+
+    service = mkOption {
+      type = types.attrsOf types.anything;
+      default =
+        let
+          stdConfig = service: {
+            server = "${service}-vm";
+            clientSocketPath = "/run/memsocket/${service}-client.sock";
+            serverSocketPath = service: suffix: "/run/memsocket/${service}${suffix}.sock";
+            userService = false;
+            serverConfig = {
+              runsOnVm = true;
+            };
+          };
+        in
+        {
+          gui = lib.attrsets.recursiveUpdate (stdConfig "gui") {
+            enabled = cfg.gui;
+            serverSocketPath = service: suffix: "/tmp/${service}${suffix}.sock";
+            serverConfig = {
+              userService = true;
+              systemdParams = service: suffix: {
+                wantedBy = [ "ghaf-session.target" ];
+                after = [ "waypipe${builtins.replaceStrings [ "-vm" ] [ "" ] suffix}.service" ];
+                partOf = [ "waypipe${builtins.replaceStrings [ "-vm" ] [ "" ] suffix}.service" ];
+                bindsTo = [ "waypipe${builtins.replaceStrings [ "-vm" ] [ "" ] suffix}.service" ];
+                serviceConfig = {
+                  KillSignal = "SIGTERM";
+                  ExecStop = "${pkgs.coreutils}/bin/rm -f ${cfg.service.gui.serverSocketPath service suffix}";
+                };
+              };
+              multiProcess = true;
+            };
+            clients = [
+              "chrome-vm"
+              "gala-vm"
+              "zathura-vm"
+              "comms-vm"
+              "business-vm"
```

Comment:
this should be set in the modules/reference/profiles/mvp. because it it not known here in common that these are the enabled vms.